qemu/qemu.go

```diff
@@ -1755,46 +1758,34 @@ func (config *Config) appendKernel() {
 }
 
 func (config *Config) appendMemoryKnobs() {
-	if config.Knobs.HugePages {
-		if config.Memory.Size != "" {
-			dimmName := "dimm1"
-			objMemParam := "memory-backend-file,id=" + dimmName + ",size=" + config.Memory.Size + ",mem-path=/dev/hugepages,share=on,prealloc=on"
-			numaMemParam := "node,memdev=" + dimmName
-
-			config.qemuParams = append(config.qemuParams, "-object")
-			config.qemuParams = append(config.qemuParams, objMemParam)
-
-			config.qemuParams = append(config.qemuParams, "-numa")
-			config.qemuParams = append(config.qemuParams, numaMemParam)
-		}
-	} else if config.Knobs.MemPrealloc {
-		if config.Memory.Size != "" {
-			dimmName := "dimm1"
-			objMemParam := "memory-backend-ram,id=" + dimmName + ",size=" + config.Memory.Size + ",prealloc=on"
-			numaMemParam := "node,memdev=" + dimmName
-
-			config.qemuParams = append(config.qemuParams, "-object")
-			config.qemuParams = append(config.qemuParams, objMemParam)
-
-			config.qemuParams = append(config.qemuParams, "-numa")
-			config.qemuParams = append(config.qemuParams, numaMemParam)
-		}
-	} else if config.Knobs.FileBackedMem {
-		if config.Memory.Size != "" && config.Memory.Path != "" {
-			dimmName := "dimm1"
-			objMemParam := "memory-backend-file,id=" + dimmName + ",size=" + config.Memory.Size + ",mem-path=" + config.Memory.Path
-			if config.Knobs.FileBackedMemShared {
-				objMemParam += ",share=on"
-			}
-			numaMemParam := "node,memdev=" + dimmName
+	if config.Memory.Size == "" {
+		return
+	}
 
-			config.qemuParams = append(config.qemuParams, "-object")
-			config.qemuParams = append(config.qemuParams, objMemParam)
+	var objMemParam, numaMemParam string
+	dimmName := "dimm1"
+	if config.Knobs.HugePages {
+		objMemParam = "memory-backend-file,id=" + dimmName + ",size=" + config.Memory.Size + ",mem-path=/dev/hugepages"
+		numaMemParam = "node,memdev=" + dimmName
+	} else if config.Knobs.FileBackedMem && config.Memory.Path != "" {
+		objMemParam = "memory-backend-file,id=" + dimmName + ",size=" + config.Memory.Size + ",mem-path=" + config.Memory.Path
+		numaMemParam = "node,memdev=" + dimmName
+	} else {
+		objMemParam = "memory-backend-ram,id=" + dimmName + ",size=" + config.Memory.Size
+		numaMemParam = "node,memdev=" + dimmName
+	}
 
-			config.qemuParams = append(config.qemuParams, "-numa")
-			config.qemuParams = append(config.qemuParams, numaMemParam)
-		}
+	if config.Knobs.MemShared {
+		objMemParam += ",share=on"
 	}
+	if config.Knobs.MemPrealloc {
+		objMemParam += ",prealloc=on"
+	}
+	config.qemuParams = append(config.qemuParams, "-object")
+	config.qemuParams = append(config.qemuParams, objMemParam)
+
+	config.qemuParams = append(config.qemuParams, "-numa")
```

Comment:
this change is to support qemu 4.1, right?

Reply:
No, I didn't change the `config.qemuParams = append(config.qemuParams, "-numa")` part. It's present in the old code as well.

Reply:
As far as I can tell the commit is just factorizing the code.  The main change I see is that prealloc and share and be configured independently of HugePages and FileBackedMem

Reply:
@markdryan Yes, you are right.